tests/functional_tests/0003_create_test.py

```diff
@@ -21,30 +21,235 @@
+    u_flag = str(uuid.uuid4())
+
+    invoke_cli(
+        ['create', '-r', release, '-u', u_flag]
+    )
+
+    jails[u_flag] = {'release': release}
+
+
+@require_root
+@require_zpool
+def test_02_create_jail_with_name(release, jails, invoke_cli):
+    invoke_cli(
+        ['create', '-r', release, '-n', 'name_jail']
+    )
+
+    jails['name_jail'] = {'release': release}
+
+
+@require_root
+@require_zpool
+def test_03_create_jail_with_uuid_and_short(release, jails, invoke_cli):
+    u_flag = str(uuid.uuid4())
+
+    invoke_cli(
+        ['create', '-r', release, '-s', '-u', u_flag]
+    )
+
+    jails[u_flag[:8]] = {'release': release}
+
+
+@require_root
+@require_zpool
+def test_04_create_count_jails_with_name(release, jails, invoke_cli):
+    count = 3
+
+    invoke_cli(
+        ['create', '-r', release, '-n', 'jail', '-c', count]
+    )
+
+    for i in range(1, count + 1):
+        jails[f'jail_{i}'] = {'release': release, 'start': False}
+
+
+@require_root
+@require_zpool
+def test_05_create_thick_config_jail(release, jails, invoke_cli):
+    invoke_cli(
+        ['create', '-r', release, '-n', 'thick_config', '-C']
+    )
+
+    jails['thick_config'] = {'release': release, 'type': 'thick_config'}
+
+
+@require_root
+@require_zpool
+def test_06_verify_thick_config_jail_props(jails, config_data):
+    if 'thick_config' not in jails:
+        pytest.skip('Skipping verify thick config jail props test')
+
+    data = config_data('thick_config') or {}
+
+    assert 'CONFIG_TYPE' in data,\
+        'Thick config jail does not has all properties'
+
+
+@require_root
+@require_zpool
+def test_07_create_empty_jail(jails, invoke_cli):
+    invoke_cli(
+        ['create', '-e', '-n', 'empty_jail']
+    )
+
+    jails['empty_jail'] = {'release': 'EMPTY', 'start': False, 'type': 'empty'}
+
+
+@require_root
+@require_zpool
+def test_08_create_base_jail(release, jails, invoke_cli):
+    invoke_cli(
+        ['create', '-r', release, '-n', 'base_jail', '-b']
+    )
+
+    jails['base_jail'] = {'release': release, 'type': 'base'}
+
+
+@require_root
+@require_zpool
+def test_09_create_thick_jail(release, jails, invoke_cli):
+    invoke_cli(
+        ['create', '-r', release, '-n', 'thick_jail', '-T']
+    )
+
+    jails['thick_jail'] = {'release': release, 'type': 'thick'}
+
+
+@require_root
+@require_zpool
+@require_dhcp
+def test_10_test_dhcp_connectivity_in_jail(
+        release, jails, invoke_cli, run_cmd
+):
+    invoke_cli(
+        ['create', '-r', release, '-n', 'dhcp_jail', 'dhcp=on'],
+        'Failed to create DHCP Jail'
+    )
+
+    jails['dhcp_jail'] = {
+        'release': release, 'networking': False, 'start': False
+    }
+
+    # Let's start it and test it's connectivity
+    # Starting can has it's own issues - but that would at least tell us
+    # that we don't have connectivity in a dhcp created jail
+
+    invoke_cli(
+        ['start', 'dhcp_jail'],
+        'Failed to start DHCP Jail'
+    )
+
+    try:
+        addr = run_cmd(
```

Comment:
This may have implications on older RELEASEs in terms of split. We should copy iocage here. Or simply catch the exception from iocage instead of duping the code and worrying about chasing. We can change iocage to throw a specific exception for this instead that we can catch here.